data/mods/ssb/abilities.ts

```diff
@@ -150,6 +150,60 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 		},
 	},
 
+// fart
+	bipolar: {
+		desc: "When this Pokemon switches in, it changes to two random types and gets STAB.",
+		shortDesc: "2 random types + STAB on switch-in.",
+		name: "Bipolar",
+		onSwitchIn(pokemon) {
+			const typeMap = {
+				Normal: "Return",
+				Fighting: "Sacred Sword",
+				Flying: "Drill Peck",
+				Poison: "Poison Jab",
+				Ground: "Earthquake",
+				Rock: "Stone Edge",
+				Bug: "Lunge",
+				Ghost: "Shadow Bone",
+				Steel: "Iron Head",
+				Electric: "Zing Zap",
+				Psychic: "Psychic Fangs",
+				Ice: "Icicle Crash",
+				Dragon: "Dual Chop",
+				Dark: "Jaw Lock",
+				Fairy: "Play Rough",
+			};
+			let types = Object.keys(typeMap);
+			this.prng.shuffle(types);
+			let newTypes = [types[0], types[1]];
+			this.add('-start', pokemon, 'typechange', newTypes.join('/'));
+			pokemon.setType(newTypes);
+			let move = this.dex.getMove(typeMap[newTypes[0]]);
+			pokemon.moveSlots[3] = pokemon.moveSlots[1];
+			pokemon.moveSlots[1] = {
+					move: move.name,
+					id: move.id,
+					pp: move.pp,
+					maxpp: move.pp,
+					target: move.target,
+					disabled: false,
+					used: false,
+					virtual: true,
+			};
+			move = this.dex.getMove(typeMap[newTypes[1]]);
+			pokemon.moveSlots[2] = {
```

Comment:
afaik, you'll be overwriting your custom move

Reply:
that's correct, the custom move is in slot 2.
that's what line 182 (pokemon.moveSlots[3] = pokemon.moveSlots[1];) is for, to put the custom move at the end

Reply:
would it be better to put placeholder moves in slot 1 & 2 and overwrite them?

Reply:
ah,
I think you can cut down on hacking by using the unshift array method (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift).

Reply:
if it doesnt work i can just merge as is